packages/liveblocks-react-ui/src/components/Composer.tsx

```diff
@@ -555,15 +594,24 @@ export const Composer = forwardRef(
     forwardedRef: ForwardedRef<HTMLFormElement>
   ) => {
     const client = useClient();
-    const createThread = useCreateThread();
-    const createComment = useCreateComment();
-    const editComment = useEditComment();
     const { preventUnsavedComposerChanges } = useLiveblocksUIConfig();
     const hasResolveMentionSuggestions =
       client[kInternal].resolveMentionSuggestions !== undefined;
     const isEmptyRef = useRef(true);
     const isEmojiPickerOpenRef = useRef(false);
     const $ = useOverrides(overrides);
+    const permissions = useRoomPermissions(useComposerRoomId());
+    const canComment =
+      permissions.has(Permission.CommentsWrite) ||
+      permissions.has(Permission.Write);
```

Comment:
I introduced a new internal/private hook called `useRoomPermissions` that returns the set of permissions for the provided room. We use this set of permissions to check the composer should be disabled or not.